From 041027724a7f3b36f8384ab57e2385c4ea796013 Mon Sep 17 00:00:00 2001
From: Shizhao Liu <lshizhao@amazon.com>
Date: Tue, 3 Feb 2026 20:22:53 -0800
Subject: [PATCH 17/18] Fix containerd sha256 checksum url

(1) Use containerd_sha256 value for containerd checksum.
(2) Containerd files should be unpack from root directory
instead of /usr/local
(3) Fix ctr binary path when loading kubernete images.

Signed-off-by: Shizhao Liu <lshizhao@amazon.com>
---
 images/capi/ansible/roles/containerd/tasks/main.yml | 4 ++--
 images/capi/ansible/roles/kubernetes/tasks/url.yml  | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/images/capi/ansible/roles/containerd/tasks/main.yml b/images/capi/ansible/roles/containerd/tasks/main.yml
index f2894123f..51a8bb10a 100644
--- a/images/capi/ansible/roles/containerd/tasks/main.yml
+++ b/images/capi/ansible/roles/containerd/tasks/main.yml
@@ -26,7 +26,7 @@
 - name: Download containerd
   ansible.builtin.get_url:
     url: "{{ containerd_url }}"
-    checksum: sha256:{{ containerd_url }}.sha256sum
+    checksum: sha256:{{ containerd_sha256 }}
     dest: /tmp/{{ containerd_filename }}
     mode: "0600"
 
@@ -67,7 +67,7 @@
   ansible.builtin.unarchive:
     remote_src: true
     src: /tmp/{{ containerd_filename }}
-    dest: "{{ containerd_prefix | default('/usr/local') }}"
+    dest: "{{ containerd_prefix | default('/') }}"
     extra_opts:
       - --no-overwrite-dir
   when: ansible_os_family != "Flatcar"
diff --git a/images/capi/ansible/roles/kubernetes/tasks/url.yml b/images/capi/ansible/roles/kubernetes/tasks/url.yml
index 465337235..f1ca3e925 100644
--- a/images/capi/ansible/roles/kubernetes/tasks/url.yml
+++ b/images/capi/ansible/roles/kubernetes/tasks/url.yml
@@ -118,7 +118,7 @@
     path: /tmp/modify-k8s-img.sh
 
 - name: Load Kubernetes images
-  ansible.builtin.shell: CONTAINERD_NAMESPACE="k8s.io" {{ sysusr_prefix }}/bin/ctr --address={{ containerd_cri_socket }} images import /tmp/{{ item }}
+  ansible.builtin.shell: CONTAINERD_NAMESPACE="k8s.io" {{ sysusr_prefix }}/local/bin/ctr --address={{ containerd_cri_socket }} images import /tmp/{{ item }}
   loop: "{{ kubernetes_imgs }}"
 
 - name: Remove Kubernetes images
-- 
2.50.1 (Apple Git-155)

